synthetic/pdf/parser.py

```diff
@@ -288,7 +311,7 @@ def parse_pdf(
     )
 
     try:
-        synthesize_fn(pdf_file)
+        timeout(synthesize_fn, args=(pdf_file,), timeout_in_seconds=timeout_in_seconds)
```

Comment:
Could we consider this package? https://github.com/pnpnpn/timeout-decorator

Reply:
Yes we should consider using a package for this. The one you linked hasn't been updated in 2 years and is still marked as in beta. I'll make an issue and we can look into this later.